File: trunk/SuperShooter/Assets/MainMenu/MainMenuGUI.js
var customSkin : GUISkin;
var isMenuShown : boolean = false;
var cameraToMask : Camera;
var player : GameObject;

private var HIDE_GUI = 0x7FFFFFFF;
private var SHOW_GUI = 0x80000000;

function OnGUI() {
	var e = Event.current;
	if (e){
		switch(e.type){
			case EventType.KeyDown:
				switch(e.keyCode){
					case KeyCode.Escape:
						isMenuShown = !isMenuShown;	
						break;
				}
				break;
		}	
	}
	

		DoLayout();
		
		
	Screen.showCursor=isMenuShown;
	
	if (cameraToMask)
		cameraToMask.cullingMask = isMenuShown ? (cameraToMask.cullingMask & HIDE_GUI) : (cameraToMask.cullingMask | SHOW_GUI);
	
	if (player){
		for (var fpsWalker in player.GetComponentsInChildren(FPSWalker))
			fpsWalker.enabled = !isMenuShown;
		for (var mouseLook in player.GetComponentsInChildren(MouseLook))
			mouseLook.enabled = !isMenuShown;
		for (var gunTrigger in player.GetComponentsInChildren(GunTrigger))
			gunTrigger.enabled = !isMenuShown;
	}
	
}

function DoLayout(){


	var guiH = 600;
	var guiW = 500;

	
	GUI.skin = customSkin;
	if (isMenuShown ){
		
		GUILayout.BeginArea(Rect(Screen.width / 2 - guiW/2, Screen.height / 2 - guiH/2 + 80, guiW, guiH));
	
		GUILayout.BeginVertical (customSkin.box);
		GUILayout.Label("Menu");
		GUILayout.Space(50);
		
		if (Application.loadedLevel != 1 && Application.loadedLevel != 2 && Application.loadedLevel != 3){
			var skirmish = GUILayout.Button("Practice");
			var campaign = GUILayout.Button("Campaign");
			var campaign2 = GUILayout.Button("Campaign2");
		}
		GUILayout.FlexibleSpace();
		
		if (Application.loadedLevel != 0)
			var leave = GUILayout.Button("Leave Game");
		var exit = GUILayout.Button("Exit");
		GUILayout.EndVertical();
		GUILayout.EndArea ();
	
		if (exit){
			Application.Quit();
		}else if (leave){
			Application.LoadLevel(0);	
		}else if (skirmish) {
			Application.LoadLevel(1);
		}else if (campaign){
			Application.LoadLevel(2);
		}else if (campaign2){
			Application.LoadLevel(3);			
		}
		
	}
}

File: SuperShooter/Assets/MainMenu/MainMenuGUI.cs
using UnityEngine;
using System.Collections;
using UnityEngine.SceneManagement;

[System.Serializable]
public partial class MainMenuGUI : MonoBehaviour
{
    public GUISkin customSkin;
    public bool isMenuShown;
    public Camera cameraToMask;
    public GameObject player;
    private const int HIDE_GUI = 2147483646;
    private const int SHOW_GUI = 2147483647;

    public virtual void OnGUI()
    {
        Event e = Event.current;
        if (e != null)
        {
            switch (e.type)
            {
                case EventType.KeyDown:
                    switch (e.keyCode)
                    {
                        case KeyCode.Escape:
                            this.isMenuShown = !this.isMenuShown;
                            break;
                    }

                    break;
            }
        }

        this.DoLayout();
        Cursor.visible = this.isMenuShown;
        if (this.cameraToMask)
        {
            this.cameraToMask.cullingMask = this.isMenuShown
                ? this.cameraToMask.cullingMask & HIDE_GUI
                : this.cameraToMask.cullingMask | SHOW_GUI;
        }

        if (this.player)
        {
            foreach (Component fpsWalker in this.player.GetComponentsInChildren(typeof(FPSWalker)))
            {
                ((Behaviour) fpsWalker).enabled = !this.isMenuShown;
            }

            foreach (Component mouseLook in this.player.GetComponentsInChildren(typeof(MouseLook)))
            {
                ((Behaviour) mouseLook).enabled = !this.isMenuShown;
            }

            foreach (Component gunTrigger in this.player.GetComponentsInChildren(typeof(GunTrigger)))
            {
                ((Behaviour) gunTrigger).enabled = !this.isMenuShown;
            }
        }
    }

    public virtual void DoLayout()
    {
        int guiH = 600;
        int guiW = 500;
        GUI.skin = this.customSkin;
        if (this.isMenuShown)
        {
            GUILayout.BeginArea(new Rect((Screen.width / 2) - (guiW / 2), ((Screen.height / 2) - (guiH / 2)) + 80, guiW,
                guiH));
            GUILayout.BeginVertical(this.customSkin.box, new GUILayoutOption[] { });
            GUILayout.Label("Menu", new GUILayoutOption[] { });
            GUILayout.Space(50);

            bool skirmish = false;
            bool campaign = false;
            bool campaign2 = false;
            bool exit = false;
            bool leave = false;

            int activeSceneBuildIndex = SceneManager.GetActiveScene().buildIndex;
            if (activeSceneBuildIndex != 1 && activeSceneBuildIndex != 2 && activeSceneBuildIndex != 3)
            {
                skirmish = GUILayout.Button("Practice", new GUILayoutOption[] { });
                campaign = GUILayout.Button("Campaign", new GUILayoutOption[] { });
                campaign2 = GUILayout.Button("Campaign2", new GUILayoutOption[] { });
            }

            GUILayout.FlexibleSpace();
            if (activeSceneBuildIndex != 0)
            {
                leave = GUILayout.Button("Leave Game", new GUILayoutOption[] { });
            }

            exit = GUILayout.Button("Exit", new GUILayoutOption[] { });
            GUILayout.EndVertical();
            GUILayout.EndArea();
            if (exit)
            {
                Application.Quit();
            }
            else
            {
                if (leave)
                {
                    SceneManager.LoadScene(0);
                    // Application.LoadLevel(0);
                }
                else
                {
                    if (skirmish)
                    {
                        SceneManager.LoadScene(1);
                        //Application.LoadLevel(1);
                    }
                    else
                    {
                        if (campaign)
                        {
                            SceneManager.LoadScene(2);
                            // Application.LoadLevel(2);
                        }
                        else
                        {
                            if (campaign2)
                            {
                                SceneManager.LoadScene(3);
                                // Application.LoadLevel(3);
                            }
                        }
                    }
                }
            }
        }
    }
}